Infomail system iteration 2

OttoRoming opened the pull request:
Lägger till frontend för mailsystemet

<img width="1721" height="1118" alt="Screenshot 2025-12-08 at 20-01-22 Mailutskick – Medlem D-sektionen" src="https://github.com/user-attachments/assets/34787d3e-6832-4dc8-8803-edb613c3a49f" />


Korrekt felhantering mot api görs av både preview och skickning
<img width="1531" height="860" alt="Screenshot 2025-12-08 at 21-01-14 Mailutskick – Medlem D-sektionen" src="https://github.com/user-attachments/assets/b9256c4d-9663-4f9c-819c-80a73bc4e9f0" />
<img width="1531" height="860" alt="Screenshot 2025-12-08 at 21-01-53 Mailutskick – Medlem D-sektionen" src="https://github.com/user-attachments/assets/3e629819-ba15-4f92-8bf6-0379a13babd4" />


Se denna pull request på backenden https://github.com/d-sektionen/backend/pull/96

OttoRoming committed:
functional mail sending system

OttoRoming committed:
fixed formatting issue

OttoRoming committed:
only show infomail page to infomail senders

OttoRoming committed:
improved preview

OttoRoming committed:
fix info chief typo

OttoRoming committed:
reset the error message to null on success to remove the error message

OttoRoming committed:
added loading spinner when mail preview is loading

OttoRoming committed:
use unicode horizontal-ellipsis

OttoRoming committed:
add raw html mode to the infomail system

OttoRoming committed:
Merge branch 'master' into send-mail-otto

OttoRoming committed:
removed unused imports

OttoRoming committed:
make if statement readable

OttoRoming committed:
remove old comments

OttoRoming committed:
fix formatting

OttoRoming committed:
fix formatting

OttoRoming committed:
use sass variables for colors

OttoRoming committed:
add a missing br and fix formatting

OttoRoming committed:
use sass variables for richtext

OttoRoming committed:
fix scss import

OttoRoming committed:
change to const

OttoRoming committed:
make mailpreview sandboxed

OttoRoming committed:
remvoe pageContext from props

OttoRoming committed:
use strong inequality for modalstate.closed

OttoRoming committed:
remove windowTitle css class import

OttoRoming committed:
remove unused import

OttoRoming committed:
use selfclosing divs

OttoRoming committed:
use scss variables for window component

OttoRoming committed:
rename textArea to TextArea

OttoRoming committed:
fix a11y

OttoRoming committed:
remove fixme comment

YouKnowBlom reviewed (changes requested):
Väldigt rimliga ändringsförslag från Copiloten tyvärr

OttoRoming committed:
Merge branch 'master' of github.com:d-sektionen/medlem into send-mail-otto

OttoRoming committed:
idk even know

Kesomannen commented on src/components/mail/page.js:
```diff
@@ -0,0 +1,188 @@
+  const [infoChiefContent, setInfoChiefContent] = useState('')
+  const [content, setContent] = useState('')
+  const [rawMode, setRawMode] = useState(false)
+  const [modalState, setModalState] = useState(ModalState.CLOSED)
+
+  function sendMail() {
+    setModalState(ModalState.SENDING)
+    post('/mail/send/', { subject, content, infoChiefContent })
+      .then(() => {
+        setModalState(ModalState.SUCCESS)
+      })
+      .catch(() => {
+        setModalState(ModalState.FAIL)
+      })
+  }
+
+  function sendMailToMyself() {
+    setModalState(ModalState.SENDING)
+    post('/mail/send-self/', { subject, content, infoChiefContent })
+      .then(() => {
+        setModalState(ModalState.SUCCESS)
+      })
+      .catch(() => {
+        setModalState(ModalState.FAIL)
+      })
```
this is valid

Kesomannen replied:
@copilot apply changes based on [this feedback](https://github.com/d-sektionen/medlem/pull/94#discussion_r2907106084)

OttoRoming committed:
add error message in fail modal

OttoRoming committed:
use more descriptive fail messages

OttoRoming committed:
nevermind

Kesomannen commented on src/components/mail/preview.js:
```diff
@@ -0,0 +1,83 @@
+  const [isLoading, setIsLoading] = useState(true)
+  const [errorMessage, setErrorMessage] = useState(null)
+
+  useEffect(() => {
+    setIsLoading(true)
+    const controller = new AbortController()
+    const signal = controller.signal
+    let isLatest = true
+
+    post('/mail/preview/', { content, infoChiefContent }, { signal })
+      .then((data) => {
+        if (!isLatest) return // ignore stale response
+        setPreview(data.data)
+        setErrorMessage(null)
+      })
```
jag håller med

Copilot committed:
Improve error handling in sendMail/sendMailToMyself to show backend error details

Agent-Logs-Url: https://github.com/d-sektionen/medlem/sessions/b9a78c77-c504-4478-b46c-0b4efa6c2a9a

Co-authored-by: Kesomannen <113015915+Kesomannen@users.noreply.github.com>

Copilot committed:
Extract error formatting into shared helper function

Agent-Logs-Url: https://github.com/d-sektionen/medlem/sessions/b9a78c77-c504-4478-b46c-0b4efa6c2a9a

Co-authored-by: Kesomannen <113015915+Kesomannen@users.noreply.github.com>
